enhancements/network/ingress-node/ingress-node-firewall-with-ebpf.md

```diff
@@ -0,0 +1,480 @@
+}
+
+// ProtocolType defines the protocol types that are supported
+type IngressNodeFirewallRuleProtocolType string
+
+const (
+	// ProtocolTypeICMP refers to the ICMP protocol
+	ProtocolTypeICMP ProtocolType = "ICMP"
+
+	// ProtocolTypeICMPv6 refers to the ICMPv6 protocol
+	ProtocolTypeICMPv6 ProtocolType = "ICMPv6"
+
+	// ProtocolTypeTCP refers to the TCP protocol, for either IPv4 or IPv6
+	ProtocolTypeTCP ProtocolType = "TCP"
+
+	// ProtocolTypeUDP refers to the UDP protocol, for either IPv4 or IPv6
+	ProtocolTypeUDP ProtocolType = "UDP"
+
+	// ProtocolTypeSCTP refers to the SCTP protocol, for either IPv4 or IPv6
+	ProtocolTypeSCTP ProtocolType = "SCTP"
+)
+
+// IngressNodeFirewallActionType indicates whether an IngressNodeFirewallRule allows or denies traffic
+// +kubebuilder:validation:Pattern=`^Allow|Deny$`
+type IngressNodeFirewallActionType string
+
+const (
+	IngressNodeFirewallAllow IngressNodeFirewallActionType = "Allow"
+	DIngressNodeFirewallDeny IngressNodeFirewallActionType = "Deny"
+)
+
+// IngressNodeFirewallRules define ingress node firewall rule
+type IngressNodeFirewallRules struct {
+// FromCIDRS is A list of CIDR from which we apply node firewall rule
+FromCIDRs []IngressNodeFirewallFromCIDR `json:"fromCIDRs" protobuf:"bytes,1,opt,name=fromCIDRs"`
+// FirewallProtocolRules is A list of per protocol ingress node firewall rules
+FirewallProtocolRules []IngressNodeFirewallProtocolRule `json:"firewallProtocolRules" protobuf:"bytes,2,opt,names=firewallProtocolRules"`
+}
+
+// NodeIngressFirewallConfigConfigSpec consists of a node's ingress firewall rules
+// +k8s:openapi-gen=true
+type NodeIngressFirewallConfigSpec struct {
+	// A list of ingress firewall policy rules.
+	// empty list indicates no ingress firewall i.e allow all incoming traffic.
+	// +kubebuilder:validation:Optional
+	// +optional
+	Ingress []IngressNodeFirewallRules `json:"ingress" protobuf:"bytes,2,opt,name=ingress"`
+}
+
+
+// NodeEndPointConfigSpec consists of list of interfaces where ingress node firewall policy will be attached to
+// +k8s:openapi-gen=true
+type NodeEndPointConfigSpec struct {
+	// A list of interfaces where the ingress firewall policy will be applied on.
+	// empty list indicates the firewall policy applied on all interfaces
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:MinItems:=1
+	// +kubebuilder:validation:UniqueItems:=true
+	// +optional
+    // +nullable
+	Interface *[]string `json:"interface" protobuf:"bytes,1,opt,name=interface"`
+}
+
+```
+
+### Test Plan
+
+- Unit tests coverage
+- IPv4/IPv6 E2E coverage to validate ingress node firewall functionality on OCP cluster with multiple external traffic streams to match allow and deny actions and verify statistics per rule.
+
+### Risks and Mitigations
+
+admin could leave the cluster in unusable state by configuring bad rules to block traffic to api server or drop DNS,
```

Comment:
EBPF's effects on other core dataplane components seems like a fairly lager risk here as well

Reply:
why would there be any impact on core dataplane with eBPF ?

Reply:
Since we're accessing data before iptables and other datapaths we could break certain expected traffic patterns (I know this is just a firewall drop/allow but using EBPF for traffic verdicts in general is a risk in my opinion)

Reply:
till we try out we won't know we minimize the risk by using fail open i.e we allow if we don't know so we will have no impact to traffic patterns we didn't anticipate